cpp/inc/bond/ext/grpc/client_callback.h

```diff
@@ -11,50 +11,56 @@
 #include <grpcpp/client_context.h>
 
 #include <memory>
-#include <utility>
+
 
 namespace bond { namespace ext { namespace gRPC {
 
     /// @brief The client-side results of a unary call.
-    template <typename TResponse>
-    struct unary_call_result
+    template <typename Response>
+    class unary_call_result
     {
-        /// @brief The response received from the service.
-        ///
-        /// @note Depending on the implementation of the service, this may or
-        /// may not contain an actual response. Consult the documentation for
-        /// the service to determine under what conditions it sends back a
-        /// response.
-        bond::bonded<TResponse> response;
-        /// @brief The status of the request.
-        grpc::Status status;
-        /// @brief The client context underwhich the request was executed.
-        std::shared_ptr<grpc::ClientContext> context;
-
-        /// @brief Create a unary_call_result with the given context and
-        /// empty \ref response and \ref status members.
-        ///
-        /// @param context the context underwhich the request is being
-        /// executed.
-        explicit unary_call_result(std::shared_ptr<grpc::ClientContext> context)
-            : response(),
-              status(),
-              context(std::move(context))
-        { }
-
+    public:
         /// @brief Create a unary_call_result with the given values.
         ///
         /// @param response The response.
         /// @param status The status.
-        /// @param context the context underwhich the request is being executed.
+        /// @param context the context under which the request is being executed.
         unary_call_result(
-            const bond::bonded<TResponse>& response,
+            bonded<Response> response,
             const grpc::Status& status,
             std::shared_ptr<grpc::ClientContext> context)
-            : response(response),
-              status(status),
-              context(std::move(context))
+            : _response(std::move(response)),
+              _status(status),
+              _context(std::move(context))
         { }
+
+        const bonded<Response>& response() const BOND_NOEXCEPT
+        {
+            return _response;
+        }
+
+        const grpc::Status& status() const BOND_NOEXCEPT
+        {
+            return _status;
+        }
+
+        const std::shared_ptr<grpc::ClientContext>& context() const BOND_NOEXCEPT
+        {
+            return _context;
+        }
+
+    private:
+        /// @brief The response received from the service.
```

Comment:
These comments should be on the new accessor methods instead of the now-private members.